input/pickle.go

```diff
@@ -60,8 +60,17 @@ func (p *Pickle) Handle(c io.Reader) error {
 			return fmt.Errorf("couldn't read payload prefix: %s", err.Error())
 		}
 
-		// payload must start with opProto, <version>, opEmptyList
-		if prefix[0] != '\x80' || prefix[2] != ']' {
+		// payload must start with:
+		switch true {
+		// version 4 - opProto, <version>, opFrame
+		case prefix[0] == '\x80' && prefix[2] == '\x95':
+		// version 2, 3 - opProto, <version>, opEmptyList
+		case prefix[0] == '\x80' && prefix[2] == ']':
+		// version 1 - opEmptyList
+		case prefix[0] == ']':
+		// version 0 - opMark, opList
+		case prefix[0] == '(' && prefix[1] == 'l':
```

Comment:
looks like v0 and v1 need fewer bytes, so the Peek(3) call above asks for more data than needed.
In practice, I don't think it matters as there should always be enough data following so that reading 3 bytes would always succeed.
but I think we should make this more precise for code clarity and to make sure the error message in the Peek error case is always accurate.

Reply:
There is no case where a valid graphite pickle stream will be shorter than 3 bytes.  Are you suggesting that we use multiple Peek calls of different lengths? That just seems like it would make things more complicated than they need to be.

Reply:
I'm also on the fence about my own suggestion,
the main reason for it, is that if for whatever reason the amount of data is less than 3 bytes (e.g. invalid input data), and we use v0 or v1, with the current code, it would return the error "couldn't read payload prefix" which is confusing and wrong.